ee/tabby-webserver/src/service/analytic.rs

```diff
@@ -0,0 +1,150 @@
+use std::sync::Arc;
+
+use async_trait::async_trait;
+use chrono::{DateTime, Utc};
+use juniper::ID;
+use tabby_db::DbConn;
+
+use super::AsRowid;
+use crate::schema::{
+    analytic::{AnalyticService, CompletionStats, Language},
+    Result,
+};
+
+struct AnalyticServiceImpl {
+    db: DbConn,
+}
+
+#[async_trait]
+impl AnalyticService for AnalyticServiceImpl {
+    // FIXME(boxbeam): Implementing in memory caching with 1 hour expiry.
+    async fn daily_stats_in_past_year(&self, users: Vec<ID>) -> Result<Vec<CompletionStats>> {
+        let users = users
+            .into_iter()
+            .filter_map(|id| id.as_rowid().ok())
+            .collect();
+        let stats = self.db.compute_daily_stats_in_past_year(users).await?;
+        let stats = stats
+            .into_iter()
+            .map(|s| CompletionStats {
```

Comment:
Implement `Into` so that this conversion is not redundant in the following method

Reply:
1 day offset is not something we will have for all such conversions (otherwise we can just have the start field). an `Into` trait is not suitable.